airbyte-cdk/python/airbyte_cdk/sources/streams/utils/stream_helper.py

```diff
@@ -36,7 +36,7 @@ def get_first_record_for_slice(stream: Stream, stream_slice: Optional[Mapping[st
     :return: StreamData containing the first record in the slice
     """
     # Store the original value of exit_on_rate_limit
-    exit_on_rate_limit_has_setter = hasattr(stream.exit_on_rate_limit, "setter")
```

Comment:
why not add the method to the interface?

Reply:
updated